### ISSUE TITLE
Bug - [react-table] - Using a Table component causes all of PatternFly's CSS to be imported

### ISSUE THREAD

**Issue by jelly** [patternfly/patternfly-react#10618]
**Describe the problem**

We have been using esbuild for a while now with Cockpit, and recently we found that it includes a [bundle analyzer](https://esbuild.github.io/analyze/). Our bundle is great large (uncompressed) so it would be interesting to see if we can make it smaller.

(If you want to see this in action, upload this [meta.json](https://github.com/user-attachments/files/15863491/meta.json) file to the esbuild bundle analyzer).

What stands out is the 1.5 mb of react-styles which are imported, this project (cockpit-machines) does not use a `Masthead` component but the css is still imported.

![image](https://github.com/patternfly/patternfly-react/assets/67428/0df521ce-0d9f-4da3-90ea-e23e59798a02)

The bundle analyzer can also show **why** something is imported

![image](https://github.com/patternfly/patternfly-react/assets/67428/2edb7ee7-b6e8-41ec-988a-764af7c752f1)

So importing `Table/index.js` imports

```
import { useOUIAProps, handleArrows, setTabIndex } from '@patternfly/react-core';
```

The problem here is that `@patternfly/react-core` is a [barrel file](https://marvinh.dev/blog/speeding-up-javascript-ecosystem-part-7/), so importing it this way imports `@patternfly/react-core/dist/js/index.js`:

```
[jelle@t14s][~/projects/cockpit-machines]%cat -p node_modules/@patternfly/react-core/dist/esm/index.js
export * from './components';
export * from './layouts';
export * from './helpers';
export * from './styles';
//# sourceMappingURL=index.js.map
```

So basically it imports all of PatternFly, esbuild seems to be smart enough to not include unused JavaScript but it seems it doesn't handle CSS.

An easy fix would be for `react-table` to import for example `setTabIndex` from `react-core/dist/js/helpers/KeyboardHandler.js` directly.

**Comment by martinpitt** [comment]
In fact we do this a lot in our projects: We never import the full `@patternfly/react-core'`, but only the components which we need, like in https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/networks/network.jsx:
```js
import { Button } from "@patternfly/react-core/dist/esm/components/Button";
import { DropdownItem } from "@patternfly/react-core/dist/esm/components/Dropdown";
import { Tooltip } from "@patternfly/react-core/dist/esm/components/Tooltip";
```

**Comment by jelly** [comment]
I looked at `react-table` and edited our `node_modules` to see if it would make an impact:

```
diff --git a/@patternfly/react-table/dist/esm/components/Table/Table.js b/@patternfly/react-table/dist/esm/components/Table/Table.js
index 702ab38a2d..e6c40d5ace 100644
--- a/@patternfly/react-table/dist/esm/components/Table/Table.js
+++ b/@patternfly/react-table/dist/esm/components/Table/Table.js
@@ -5,7 +5,8 @@ import stylesGrid from '@patternfly/react-styles/css/components/Table/table-grid
 import stylesTreeView from '@patternfly/react-styles/css/components/Table/table-tree-view.mjs';
 import { css } from '@patternfly/react-styles';
 import { toCamel } from './utils';
-import { useOUIAProps, handleArrows, setTabIndex } from '@patternfly/react-core';
+import { handleArrows, setTabIndex } from '@patternfly/react-core/dist/esm/helpers/KeyboardHandler';
+import { useOUIAProps } from "@patternfly/react-core/dist/esm/helpers/OUIA/ouia";
 import { TableGridBreakpoint } from './TableTypes';
 export const TableContext = React.createContext({
     registerSelectableRow: () => { }
```

Old [meta.json](https://github.com/user-attachments/files/15864421/meta.json)
New [meta.json](https://github.com/user-attachments/files/15864418/meta.json)

In numbers the result is:
8 mb input size => 4.7 mb output size
6.7 mb input size => 4.0 mb output size

This would be a small but nice reduction in final bundle size for cockpit-ostree, for cockpit-machines I would need to patch more PF components to get similar results.

Old:
![image](https://github.com/patternfly/patternfly-react/assets/67428/aac7d0cc-6f13-43d3-ba50-44dd760dd3b4)

New:
![image](https://github.com/patternfly/patternfly-react/assets/67428/ac9d4d55-c61d-4dfc-8f55-35357106a7ff)

**Comment by jelly** [comment]
Need to recheck with PF6 but this is still an issue in PF5.
